src/huggingface_hub/_commit_api.py

```diff
@@ -483,14 +493,15 @@ def fetch_upload_modes(
         preupload_info = _validate_preupload_info(resp.json())
         upload_modes.update(**{file["path"]: file["uploadMode"] for file in preupload_info["files"]})
 
+    # Set upload mode for each addition operation
+    for addition in additions:
+        addition._upload_mode = upload_modes[addition.path_in_repo]
+
     # Empty files cannot be uploaded as LFS (S3 would fail with a 501 Not Implemented)
     # => empty files are uploaded as "regular" to still allow users to commit them.
     for addition in additions:
         if addition.upload_info.size == 0:
-            path = addition.path_in_repo
-            upload_modes[path] = "regular"
-
-    return upload_modes
+            addition._upload_mode = "regular"
```

Comment:
(nit) This change is a breaking change right? If this is supposed to be an internal method, let's maybe update its name to `_fetch_upload_modes`? It's already in a private `_commit_api.py` file but accessible from `from huggingface_hub.hf_api import fetch_upload_modes`

Reply:
Yes it's definitely a breaking change. I will rename it to `_fetch_upload_modes`. As you said, it's from a private file and it's undocumented in our docs. I don't expect anyone to use it given how low-level and internal the logic is.

Reply:
(I've checked in `transformers` & cie first)